cpp/oneapi/dal/algo/kmeans/backend/gpu/infer_kernel_dpc.cpp

```diff
@@ -14,95 +14,72 @@
-        par.resultsToEvaluate = static_cast<DAAL_UINT64>(daal_kmeans::computeAssignments);
-
-        auto arr_data = row_accessor<const Float>{ data }.pull(queue);
-        const auto daal_data =
-            interop::convert_to_daal_table(queue, arr_data, row_count, column_count);
-
-        auto arr_initial_centroids =
-            row_accessor<const Float>{ input.get_model().get_centroids() }.pull(queue);
+        auto arr_data = pr::table2ndarray<Float>(queue, data, sycl::usm::alloc::device);
 
         dal::detail::check_mul_overflow(cluster_count, column_count);
-        array<Float> arr_centroids =
-            array<Float>::empty(queue, cluster_count * column_count, sycl::usm::alloc::device);
-        array<Float> arr_objective_function_value =
-            array<Float>::empty(queue, 1, sycl::usm::alloc::device);
-        array<int> arr_labels = array<int>::empty(queue, row_count, sycl::usm::alloc::device);
-        array<int> arr_iteration_count = array<int>::empty(queue, 1, sycl::usm::alloc::device);
-
-        const auto daal_initial_centroids = interop::convert_to_daal_table(queue,
-                                                                           arr_initial_centroids,
-                                                                           cluster_count,
-                                                                           column_count);
-        const auto daal_centroids =
-            interop::convert_to_daal_table(queue, arr_centroids, cluster_count, column_count);
-        const auto daal_labels = interop::convert_to_daal_table(queue, arr_labels, row_count, 1);
-        const auto daal_objective_function_value =
-            interop::convert_to_daal_table(queue, arr_objective_function_value, 1, 1);
-        const auto daal_iteration_count =
-            interop::convert_to_daal_table(queue, arr_iteration_count, 1, 1);
-
-        daal::data_management::NumericTable* daal_input[2] = { daal_data.get(),
-                                                               daal_initial_centroids.get() };
-
-        daal::data_management::NumericTable* daal_output[4] = { daal_centroids.get(),
-                                                                daal_labels.get(),
-                                                                daal_objective_function_value.get(),
-                                                                daal_iteration_count.get() };
-
-        interop::status_to_exception(
-            daal_kmeans_lloyd_dense_ucapi_kernel_t<Float>().compute(daal_input, daal_output, &par));
-
-        const auto arr_objective_function_value_host =
-            dal::backend::to_host_sync(arr_objective_function_value);
+        auto centroids_ptr =
+            row_accessor<const Float>{ input.get_model().get_centroids() }.pull(queue);
+        auto arr_centroids =
+            pr::ndarray<Float, 2>::wrap(centroids_ptr, { cluster_count, column_count });
```

Comment:
Seems like, in this case you can use `pr::table2ndarray`

Reply:
Done